lib/input_sanitizer/v2/payload_transform.rb

```diff
@@ -37,6 +38,9 @@ def has?(key)
   end
 
   def payload
-    @payload ||= original_payload.with_indifferent_access
+    # `payload` here meant not to have default proc and before rails 4.2 `.with_indifferent_access` removed it.
+    # HashWithIndifferentAccess will copy of orignal hash without default proc only on 1st level keys
+    # If value of 1st level key is a RestrictedHash as well, we need to `.as_json` to clean those defaults as well
+    @payload ||= HashWithIndifferentAccess[original_payload.as_json]
```

Comment:
Please don't use ActiveSupport's (ActiveModel ?)`.as_json` as it's known to be slow (on older rails versions even **super** slow). Use an explicit json encoder or multi_json. Do you have any hard requirements to using the activesupport version?